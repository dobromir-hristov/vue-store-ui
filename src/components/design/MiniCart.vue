<template>
  <div class="MiniCart">
    <div v-if="items.length">
      <div
        v-for="(item, index) in items"
        :key="index"
        class="flex justify-between items-center"
        data-testid="MiniCartItem"
      >
        <div>
          <a
            class="text-red"
            data-testid="removeItem"
            @click="$emit('remove', item.id)"
          >
            <v-icon
              type="times"
              size="xs"
            />
          </a>
          {{ item.name }}
        </div>
        <div>
          <v-icon type="times"/>
          {{ item.quantity }}
        </div>
      </div>
    </div>
    <div
      v-else
      data-testid="NoCartItems"
    >
      No items in cart yet
    </div>
  </div>
</template>

<script>
/**
 * @module MiniCart
 */
export default {
  name: 'MiniCart',
  props: {
    items: {
      type: Array,
      default: () => ([])
    }
  }
}
</script>
