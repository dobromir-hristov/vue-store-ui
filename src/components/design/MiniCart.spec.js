import MiniCart from './MiniCart'

const itemsFactory = () => ([
  { id: 1, name: 'Whatever', quantity: 1 },
  { id: 2, name: 'Whatever 2', quantity: 1 }
])

/**
 * @param overrides
 * @returns {{config, wrapper: Wrapper<V> | Wrapper<Vue> | *}}
 */
const createWrapper = (overrides) => {
  let config = _merge(createComponentMocks(),
    {},
    overrides
  )
  const wrapper = mount(MiniCart, config)
  return {
    config,
    wrapper
  }
}

describe('MiniCart', () => {
  it('renders the MiniCart without any items', () => {
    const { wrapper } = createWrapper()
    expect(wrapper.find(testid('NoCartItems')).exists()).toBeTruthy()
  })

  it('renders the items', () => {
    const { wrapper } = createWrapper({
      propsData: { items: itemsFactory() }
    })
    expect(wrapper.findAll(testid('MiniCartItem'))).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('emits an event with the element id', () => {
    const { wrapper } = createWrapper({
      propsData: { items: itemsFactory() }
    })
    wrapper.find(testid('removeItem')).trigger('click')
    expect(wrapper.emitted('remove')).toEqual([[1]])
  })
})
