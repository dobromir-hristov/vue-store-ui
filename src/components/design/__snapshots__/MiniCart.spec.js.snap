// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniCart renders the items 1`] = `
<div class="MiniCart">
  <div>
    <div data-testid="MiniCartItem" class="flex justify-between items-center">
      <div><a data-testid="removeItem" class="text-red"><i class="fa-times fas fa-xs fa-fw"></i></a>
        Whatever
      </div>
      <div><i class="fa-times fas fa-fw"></i>
        1
      </div>
    </div>
    <div data-testid="MiniCartItem" class="flex justify-between items-center">
      <div><a data-testid="removeItem" class="text-red"><i class="fa-times fas fa-xs fa-fw"></i></a>
        Whatever 2
      </div>
      <div><i class="fa-times fas fa-fw"></i>
        1
      </div>
    </div>
  </div>
</div>
`;
